File: ChuckNorrisIO/app/src/main/java/br/com/titoschmidt/chucknorrisio/datasource/ChuckNorrisAPI.java
package br.com.titoschmidt.chucknorrisio.datasource;

import java.util.List;

import br.com.titoschmidt.chucknorrisio.models.Joke;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ChuckNorrisAPI {

    static final String BASE_URL = "https://api.chucknorris.io/";

    @GET("jokes/categories")
    Call<List<String>> findAll();
    // o Retrofit converte o Json para a classe Joke
    @GET ("jokes/random")
    Call<Joke> findRandomBy(@Query("category") String category);
}

File: JokerApp/app/src/main/java/co/titoschmidt/tutorial/jokerappdev/data/ChuckNorrisAPI.kt
package co.titoschmidt.tutorial.jokerappdev.data

import co.titoschmidt.tutorial.jokerappdev.model.Joke
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

/*
 * 10/12/2021
 * @author titoesa@gmail.com (Tito Schmidt).
 * 
 */
interface ChuckNorrisAPI {

    @GET("jokes/categories")                                  // Retorna uma lista de strings
    fun findAllCategories(@Query("apiKey") apiKey : String) : Call<List<String>>

    @GET("jokes/random")                                                                                              // Retorna um Joke
    fun findRandom(@Query("category") categoryName: String? = null, @Query("apiKey") apiKey: String = HTTPClient.API_KEY) : Call<Joke>
}